### ISSUE TITLE
Apply a set of match rules only to new (unsaved, "Untitled") documents

### ISSUE THREAD

**Issue by Cerberus-tm** [facelessuser/ApplySyntax#133]
Great plug-in! In case I'm being stupid and this is already possible, I apologize.

An example of what I'd like to have the plug-in do is this:

1. I create a new document, without saving it, and without setting a syntax;
2. I start typing code;
3. The plug-in automatically applies the syntax "Javascript" the moment the string ".querySelector" exists within the document;
4. It then ceases trying to further apply syntaxes to that document.

Would this be practicable? (Of course "Javascript" and ".querySelector" are random examples.)

I've read the documentation and spent half an hour trying to get this to work, but any syntax only seems to be applied when I save the file.

P.S. How often does the plug-in check the contents of the file? I thought maybe upon each change in the content of the document, but perhaps that is not the case.

**Comment by facelessuser** [comment]
Sorry I didn't get to this sooner. To answer your question, it really only checks on file open and file save. There are no plans to add hooks for on_modified events etc. It adds overhead during typing that can potentially slow things down. There are ways to minimize the slow down by adding more complexity, but I honestly don't have intentions at this time to add such complexity.

**Comment by Cerberus-tm** [comment]
Oh, don't worry. It is completely understandable, if you don't have the time or are not of a mind to add this in the foreseeable future. (The only times I hate it when an issue is not taken up is when a developer suggests my idea is pointless or useless. You're nothing like that.) I have learned a bit more since I posted that issue, so perhaps I will be able to do it myself, either by editing this plug-in or by making one myself.

**Comment by facelessuser** [comment]
Yeah, the idea is fine, but it is just beyond the scope of what I am willing to maintain at this time. I support quite a number of Sublime Text plugins, so at this time, I am trying to not add additional complexity unless I have a strong desire to maintain it. This is simply a feature that I haven't personally needed, and the work is weighty enough that it isn't quit appealing at this time.

Thanks for being understanding 🙂 .

**Comment by Cerberus-tm** [comment]
Of course! I think it is sometimes a bit of a vicious circle: user suggests something; developer views it as disapproval, "why can't your programme do even this!?", and responds in a defensive or dismissive manner; user feels disparaged by god-like developer, becomes even pushier and angry, etc. And people take this experience with them in their behaviour towards the next user or the next developer. I've used several of your plug-ins, I believe, so thanks! P.S. Seeing your name in the e-mail notification, I now remember that we joked around a bit in the Teacher's Lounge. See you!
